### DISCUSSION TITLE
AMP context with Linux as Remote and FreeRTOS as Master

### DISCUSSION THREAD

**Discussion by Shivaraj199** [polarfire-soc/polarfire-soc-documentation#483] [How to]
Hello I have gone through the AMP example from Yocto in which Linux acts as MASTER and FreeRTOS acts as Remote. I would like to develop an application where FreeRTOS monitors the linux core in which linux acts as Remote, when I try to change the recipe  to execute FreeRTOS context in MASTER mode HSS gives me following error.
"e51: BEU event:              Load or store TileLink bus error".

Also adding to this question does video-kit support AMP configuration? I can see HSS picks up the AMP payload but does not show any output in UART3

**Comment by vfalanis** [comment]
Hello @Shivaraj199 ,

The standard Linux Kernel RPMSG implementation only supports Master mode, since this is not a common use case. Could you please provide some details/explanation on why you would like Linux to act as a Remote?

**Reply by Shivaraj199** [reply]
Hello @vfalanis, 
Thank you for coming back to me, 
I would like to use this configuration  so I could  command linux from FreeRTOS eg: to reboot linux context from FreeRTOS. Is this possible even if Linux acts as MASTER?

Also could you please tell if AMP is supported in video kit. As I can see uboot boots but FreeRTOS context does not. Below is the log from the video-kit

Thank you 


![log](https://github.com/user-attachments/assets/098d61e2-35a2-4c27-b1da-4822bf7968bf)

**Reply by vfalanis** [reply]
Hello @Shivaraj199 ,

The standard Linux Kernel RPMSG implementation only supports Master mode. Also, sending a reboot from a remote context to Linux is not supported.

Regarding your second question, the Icicle Kit is the board we officially use to demonstrate all AMP demos. Technically, it is possible to run AMP on the Video Kit as well, but this is something we don't provide or support in our deliverables. Customers are free to take the Icicle Kit AMP demos (including software and design configuration) as a reference and implement them on the Video Kit if desired.

Thanks
